bundles/org.eclipse.mita.program/src/org/eclipse/mita/program/linking/OperationsLinker.java

```diff
@@ -114,12 +117,44 @@ public int compare(IEObjectDescription operation1, IEObjectDescription operation
 
 	protected boolean isCallable(IEObjectDescription operation, ArgumentExpression expression) {
 		Operation op = (Operation) operation.getEObjectOrProxy();
-		List<InferenceResult> argumentTypes = getArgumentTypes(op, expression);
+
+		if (!op.eIsProxy()) {
+			return isCallableByType(operation, getArgumentTypes(op, expression));
+		} else {
+			return isCallableByName(operation, getArgumentTypes(op, expression));
+		}
```

Comment:
This is a bit ugly, but the read()/write() operations come from a library and are proxies whose parameter types cannot be resolved here. This is similar to the case in scope provider where we check super types based on names instead of the type objects itself.